examples/mnist/mnist.py

```diff
@@ -58,32 +59,47 @@ def run(train_batch_size, val_batch_size, epochs, lr, momentum, log_interval):
                                                      'nll': Loss(F.nll_loss)},
                                             device=device)
 
+    desc = "ITERATION - loss: {:.2f}"
+    pdbar = tqdm(
+        initial=log_interval, leave=False, total=len(train_loader),
+        desc=desc.format(0)
+    )
+
     @trainer.on(Events.ITERATION_COMPLETED)
     def log_training_loss(engine):
         iter = (engine.state.iteration - 1) % len(train_loader) + 1
+
         if iter % log_interval == 0:
-            print("Epoch[{}] Iteration[{}/{}] Loss: {:.2f}"
-                  "".format(engine.state.epoch, iter, len(train_loader), engine.state.output))
+            pdbar.desc = desc.format(engine.state.output)
+            pdbar.update(log_interval)
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_training_results(engine):
         evaluator.run(train_loader)
         metrics = evaluator.state.metrics
         avg_accuracy = metrics['accuracy']
         avg_nll = metrics['nll']
-        print("Training Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
-              .format(engine.state.epoch, avg_accuracy, avg_nll))
+        tqdm.write(
+            "Training Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
+            .format(engine.state.epoch, avg_accuracy, avg_nll)
+        )
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_validation_results(engine):
         evaluator.run(val_loader)
         metrics = evaluator.state.metrics
         avg_accuracy = metrics['accuracy']
         avg_nll = metrics['nll']
-        print("Validation Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
-              .format(engine.state.epoch, avg_accuracy, avg_nll))
+        tqdm.write(
+            "Validation Results - Epoch: {}  Avg accuracy: {:.2f} Avg loss: {:.2f}"
+            .format(engine.state.epoch, avg_accuracy, avg_nll))
+
+        pdbar.n = log_interval
```

Comment:
@rave78 I wonder why we need the following code ?
```
pdbar.n = log_interval
pdbar.last_print_n = log_interval
pdbar.refresh()
```
And a nit: why `pdbar` and not just `pbar` ?

Reply:
@vfdev-5 the three lines of code set the starting state for the first iteration of the next epoch.  It is not a strong argument to use `pdbar` instead of `pbar`

Reply:
I would prefer to to compact this into 
```python
pdbar.n = pdbar.last_print_n = 0
```
starting from 0 instead of `log_interval`

Reply:
I have done both changes